### PR TITLE
Resolve auto merge failing issue in choco and brew and adding check in update-brew script

### LINKED ISSUE #3179: Choco and Homebrew automation PRs failed to merge
## Bug Report

Here are the referenced GitHub actions for Choco and Homebrew:
Choco - https://github.com/vmware-tanzu/community-edition/runs/5235221353?check_suite_focus=true
Homebrew - https://github.com/vmware-tanzu/community-edition/runs/5235221350?check_suite_focus=true 

Here are the automation failed to merge Choco and Homebrew automation PRs found below:
Choco - https://github.com/vmware-tanzu/community-edition/pull/3171
Homebrew - https://github.com/vmware-tanzu/homebrew-tanzu/pull/25

PRs were merged manually when attempt continue to with the release.

## Expected Behavior

The automation (when needed) would merge the PRs automatically without human intervention.

## Environment Details

* Build version (`tanzu version`): 
* Deployment (Managed/Standalone cluster): 
* Infrastructure Provider (Docker/AWS/Azure/vSphere): 
* Operating System (client):

### PR DESCRIPTION
**Pull request by aman556**
## What this PR does / why we need it

- This PR is resolving the failing of auto-merge which required human intervention in choco and brew update scripts.
- Also it adding a part to update checksum value for choco package.
- It is adding a check in [update-homebrew-package.sh](https://github.com/vmware-tanzu/community-edition/blob/main/hack/homebrew/update-homebrew-package.sh) before merging in main.

## Details for the Release Notes (PLEASE PROVIDE)

```release-note

```

## Which issue(s) this PR fixes

Fixes: [#3179](https://github.com/vmware-tanzu/community-edition/issues/3179)

## Describe testing done for PR

Did release on my fork branch.
[Logs for above test](https://github.com/aman556/community-edition/runs/5283471379?check_suite_focus=true)

### CODE REVIEW

File: hack/choco/update-choco-package.sh
Comment (nrb) [inline hack/choco/update-choco-package.sh:63]: `""'${windows_amd64_shasum}'""` looks strange to me - should we just use `'${windows_amd64_shasum}'`? Is there something I'm missing with interpolation in the double quotes?

Comment (dvonthenen) [reply hack/choco/update-choco-package.sh:63]: or just escaping `\"` should work (i think)

File: hack/choco/update-choco-package.sh
Comment (dvonthenen) [inline hack/choco/update-choco-package.sh:69]: We dont allow merging directly into main. You must create a PR and then use the `--admin` override for the merge that PR. Please see:
https://github.com/vmware-tanzu/community-edition/blob/main/hack/release/cut-release.sh#L150-L158

Also, you shouldn't assume that you are working off of main. You need to detect which branch this commit is coming from. You can find an example here:
https://github.com/vmware-tanzu/community-edition/blob/main/hack/release/cut-release.sh#L139-L146

File: hack/homebrew/update-homebrew-package.sh
Comment (dvonthenen) [inline hack/homebrew/update-homebrew-package.sh:79]: Please see above

**Review by davidvonthenen — COMMENTED** [review]
Left some feedback

File: hack/choco/update-choco-package.sh
Comment (nrb) [reply hack/choco/update-choco-package.sh:63]: Looking at the powershell scripts, they use `'`. However, I think with discussion on #3242 means this file will be removed and replaced by a powershell script on Windows workers.

File: hack/homebrew/update-homebrew-package.sh
Comment (karuppiah7890) [reply hack/homebrew/update-homebrew-package.sh:79]: @dvonthenen About the `--admin` flag, I see `gh pr merge` mentioning this -

```
      --admin            Use administrator privileges to merge a pull request that does not meet requirements
      --auto             Automatically merge only after necessary requirements are met
```

And we want to run checks and merge the PR only after the GitHub checks pass. For Homebrew package update PR, the GitHub checks are here - https://github.com/vmware-tanzu/homebrew-tanzu/blob/main/.github/workflows/check-tce-homebrew-formula-pr.yaml

If the checks fail we want the PR to not be merged. The checks ensure that the formula works by doing a install test currently. Ideally an upgrade test should also happen as part of the checks, or we need to do the tests before we raise the PR but that's not how we do it currently

In order to run the checks and then auto merge the PR we use `--auto` and avoided using `--admin` which merges the PR without the checks passing

https://github.com/vmware-tanzu/community-edition/blob/0e10a503a4098ae45b8a1b59d80598759fac40e0/hack/homebrew/update-homebrew-package.sh#L85

Comment (karuppiah7890) [reply hack/homebrew/update-homebrew-package.sh:79]: Going by the restrictions on auto merging to main, I guess we need `--admin` and hence cannot do checks in PR, so I guess we can do checks / tests before we raise PR and then fail if checks fail, that way no PR is also created, which is also better, that way we don't have PRs that don't work. If the checks pass, we raise the PR and use `--admin` to merge the PR

File: hack/choco/update-choco-package.sh
Comment (aman556) [reply hack/choco/update-choco-package.sh:63]: Yes `\` can also work will use that.

File: hack/choco/update-choco-package.sh
Comment (aman556) [reply hack/choco/update-choco-package.sh:69]: Sure @dvonthenen I will add this thing.

File: hack/homebrew/update-homebrew-package.sh
Comment (aman556) [reply hack/homebrew/update-homebrew-package.sh:79]: > The checks ensure that the formula works by doing a install test currently.
Hii @karuppiah7890 for this I am modifying the update version script and testing before raising the PR.

Comment (aman556) [reply hack/homebrew/update-homebrew-package.sh:91]: for brew script we are using `--admin` as we are running a check before merging in the script itself.
for choco are moving with `--auto` as we have to run a PR check.

File: hack/choco/update-choco-package.sh
Comment (dvonthenen) [reply hack/choco/update-choco-package.sh:78]: Thanks. If we need rollback the change for some reason, the PR method gives us the ability to just revert the change in the PR through a click of the button.

File: hack/homebrew/update-homebrew-package.sh
Comment (dvonthenen) [reply hack/homebrew/update-homebrew-package.sh:91]: as long as the change isn't significant, `--admin` should be ok. most of the automation only bumps tiny things like version numbers. to your excellent point though, if something does go horribly wrong, creating a PR gives us an easy button to revert the commit in the PR quickly.

if `--auto` works for your change/implementation, will leave it up to you on what works best for your case. just know that it will add some delay to merge.

Comment (aman556) [reply hack/homebrew/update-homebrew-package.sh:91]: If https://github.com/vmware-tanzu/community-edition/pull/3242 got success will use `--admin` for choco too.

**Comment by aman556** [comment]
@dvonthenen can you PTAL here

File: hack/choco/update-choco-package.sh
Comment (dvonthenen) [reply hack/choco/update-choco-package.sh:78]: Still waiting for this to generate a PR instead of directly committing to the branch.

**Comment by dvonthenen** [comment]
> @dvonthenen can you PTAL here

Still waiting on this to move to generating a PR instead of committing directly into the branch:
https://github.com/vmware-tanzu/community-edition/pull/3210#discussion_r830096549

File: hack/choco/update-choco-package.sh
Comment (dvonthenen) [reply hack/choco/update-choco-package.sh:78]: We are too close to the release. Disabling this workflow until this issue is addressed and we can merge the PR.

File: hack/choco/update-choco-package.sh
Comment (dvonthenen) [inline hack/choco/update-choco-package.sh:39]: can we put a `TODO:` on this comment

File: hack/choco/update-choco-package.sh
Comment (dvonthenen) [inline hack/choco/update-choco-package.sh:53]: This assumes that we are on main. With the new support for release branches and TCE cutting GA releases from release branches.

**Review by davidvonthenen — CHANGES_REQUESTED** [review]
This also assumes we are checking out on main...
https://github.com/vmware-tanzu/community-edition/pull/3210/files#diff-3cdaa2bcf30c3e80d10e3f9d915d6fff7dff070df067e6e8f0ef52e0c5a11226R26

**Pull request closed without merging**
